Docs: Add basic query option descriptions (#74419)
grafana/grafana: docs/sources/dashboards/variables/add-template-variables/index.md

Query expressions can contain references to other variables and in effect create linked variables. Grafana detects this and automatically refreshes a variable when one of its linked variables change.

> 

Reviewer: ```suggestion

The query should return two fields named `__text` and `__value`. The `__text` column value should be unique. If it 's not unique, then the first value is used. 

```

{{% admonition type="note" %}}

> The query should return two fields named `__text` and `__value`. The `__text` column value should be unique (if it is not unique then the first value is used). Query expressions are different for each data source. For more information, refer to the documentation for your [data source]({{< relref "../../../datasources/" >}}).

Reviewer: ```suggestion
The query should return two fields named `__text` and `__value`. The `__text` column value should be unique (if it's not unique, then the first value is used). Query expressions are different for each data source. For more information, refer to the documentation for your [data source]({{< relref "../../../datasources/" >}}).
```

Reviewer: Looking at the way this admonition is being used currently on the page, it doesn't make sense to me to add this information in the same admonition. Seems like it should be its own content outside of the admonition entirely.

> The query should return two fields named `__text` and `__value`. The `__text` column value should be unique (if it is not unique then the first value is used). Query expressions are different for each data source. For more information, refer to the documentation for your [data source]({{< relref "../../../datasources/" >}}).

Reviewer: Looking at the way this admonition is being used currently on the page, it doesn't make sense to me to add this information in the same admonition. Seems like it should be its own content outside of the admonition entirely. 
```suggestion
Query expressions are different for each data source. For more information, refer to the documentation for your [data source]({{< relref "../../../datasources/" >}}).
```

   - **On Time Range Change:** Queries the data source when the dashboard time range changes. Only use this option if your variable options query contains a time range filter or is dependent on the dashboard time range.
1. In the **Query** field, enter a query.
   - The query field varies according to your data source. Some data sources have custom query editors.

>    - Make sure that the query returns values named `__text` and `__value` as appropriate in your query syntax. As a SQL-specific example, you can use a query such as `SELECT hostname AS __text, id AS __value FROM MyTable`; other query languages will vary depending on syntax.

Reviewer: I've taken out the semi-colon and just made a new sentence at the end. But that last sentence doesn't entirely make sense to me. Does my revision work?
```suggestion
   - Make sure that the query returns values named `__text` and `__value` as appropriate in your query syntax. For example, in SQL, you can use a query such as `SELECT hostname AS __text, id AS __value FROM MyTable`. Queries for other languages will vary depending on syntax.
```
